File: 999.cpp
struct Solution {
    int numRookCaptures(vector<vector<char>>& board) {
        constexpr int n = 8;
        int ri = -1, rj = -1;
        for (int i = 0; i < n; ++i){
            for (int j = 0; j < n; ++j){
                if (board[i][j] == 'R'){
                    ri = i;
                    rj = j;
                    break;
                }
            }
            if (ri != -1){
                break;
            }
        }
        int count = 0;
        for (int ii = ri - 1; ii >= 0; --ii){
            count += board[ii][rj] == 'p';
            if (board[ii][rj] != '.'){break;}
        }
        for (int ii = ri + 1; ii < n; ++ii){
            count += board[ii][rj] == 'p';
            if (board[ii][rj] != '.'){break;}
        }
        for (int jj = rj - 1; jj >= 0; --jj){
            count += board[ri][jj] == 'p';
            if (board[ri][jj] != '.'){break;}
        }
        for (int jj = rj + 1; jj < n; ++jj){
            count += board[ri][jj] == 'p';
            if (board[ri][jj] != '.'){break;}
        }
        return count;
    }
};

File: sol/999.py
class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        n = 8
        ri, rj = -1, -1
        for i in range(n):
            for j in range(n):
                if board[i][j] == 'R':
                    ri, rj = i, j
                    break
            if ri != -1:
                break
        count = 0
        for ii in range(i - 1, -1, -1):
            count += board[ii][j] == 'p'
            if board[ii][j] != '.':
                break
        for ii in range(i + 1, n):
            count += board[ii][j] == 'p'
            if board[ii][j] != '.':
                break
        for jj in range(j - 1, -1, -1):
            count += board[i][jj] == 'p'
            if board[i][jj] != '.':
                break
        for jj in range(j + 1, n):
            count += board[i][jj] == 'p'
            if board[i][jj] != '.':
                break
        return count
